File: CCF Android/CCFSGConnect/cCFSGConnect/src/main/java/com/ccfsg/connect/beans/Event.java
package com.ccfsg.connect.beans;

import java.util.ArrayList;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import android.os.Parcel;
import android.os.Parcelable;

public class Event implements Parcelable {

	private String name;
	private String subname;
	private String venue;
	private String memberPrice;
	private String guestPrice;
	private String date;
	private String startTime;
	private String endTime;
	private String verse;
	private String details;
	private String image;
	private String form;
	
	public Event( JSONObject json )
	{
		try
		{
			name = json.has( "name" ) ? json.getString( "name" ) : "";
			subname = json.has( "subname" ) ? json.getString( "subname" ) : "";
			venue = json.has( "venue" ) ? json.getString( "venue" ) : "";
			memberPrice = json.has( "memberprice" ) ? json.getString( "memberprice" ) : "";
			guestPrice = json.has( "guestprice" ) ? json.getString( "guestprice" ) : "";
			date = json.has( "date" ) ? json.getString( "date" ) : "";
			startTime = json.has( "starttime" ) ? json.getString( "starttime" ) : "";
			endTime = json.has( "endtime" ) ? json.getString( "endtime" ) : "";
			verse = json.has( "verse" ) ? json.getString( "verse" ) : "";
			details = json.has( "details" ) ? json.getString( "details" ) : "";
			image = json.has( "image" ) ? json.getString( "image" ) : "";
			form = json.has( "form" ) ? json.getString( "form" ) : "";
		}
		catch( JSONException e )
		{
			e.printStackTrace();
		}
	}
	
	@Override
	public int describeContents() {
		return 0;
	}
    
    @Override
    public void writeToParcel(Parcel out, int i) 
    {
    	out.writeString(name);
    	out.writeString(subname);
    	out.writeString(venue);
    	out.writeString(memberPrice);
    	out.writeString(guestPrice);
    	out.writeString(date);
    	out.writeString(startTime);
    	out.writeString(endTime);
    	out.writeString(verse);
    	out.writeString(details);
    	out.writeString(image);
    	out.writeString(form);
    }
	
    public Event(Parcel in)
    {
    	name = in.readString();
    	subname = in.readString();
    	venue = in.readString();
    	memberPrice = in.readString();
    	guestPrice = in.readString();
    	date = in.readString();
    	startTime = in.readString();
    	endTime = in.readString();
    	verse = in.readString();
    	details = in.readString();
    	image = in.readString();
    	form = in.readString();
    }
    
    public static final Parcelable.Creator<Event> CREATOR = new Parcelable.Creator<Event>() {
        public Event createFromParcel(Parcel in) {
            return new Event(in);
        }

        public Event[] newArray(int size) {
            return new Event[size];
        }
    };
	
	public String getName()
	{
		return name;
	}
	
	public String getSubName()
	{
		return subname;
	}
	
	public String getVenue()
	{
		return venue;
	}
	
	public String getMemberPrice()
	{
		return memberPrice;
	}

	public String getGuestPrice()
	{
		return guestPrice;
	}
	
	public String getDate()
	{
		return date;
	}
	
	public String getStartTime()
	{
		return startTime;
	}
	
	public String getEndTime()
	{
		return endTime;
	}
	
	public String getVerse()
	{
		return verse;
	}
	
	public String getDetails()
	{
		return details;
	}
	
	public String getImage()
	{
		return image;
	}
	
	public String getForm()
	{
		return form;
	}
	
	public static ArrayList<Event> toArrayList( JSONArray array )
	{
		ArrayList<Event> details = new ArrayList<Event>();
		try
		{
			for( int i = 0; i < array.length(); i++ )
			{
				Event detail = new Event( array.getJSONObject( i ) );
				details.add( detail );
			}
		}
		catch( JSONException e )
		{
			e.printStackTrace();
		}
		return details;
	}
}

File: CCF iOS/CCF/WS:Data/Event.swift
//
//  Event.swift
//  CCF
//
//  Created by Alex on 24/11/14.
//  Copyright (c) 2014 ChalkboxCreatives. All rights reserved.
//

import UIKit

class Event: NSObject {
    var name: String?
    var subname: String?
    var venue: String?
    var memberprice: String?
    var guestprice: String?
    var date: String?
    var starttime: String?
    var endtime: String?
    var verse: String?
    var details: String?
    var image: String?
    var form: String?
    
    func setData(data: NSDictionary) {
        if let name = data.valueForKey("name") as? String {
            self.name = name
        }
        if let subname = data.valueForKey("subname") as? String {
            self.subname = subname
        }
        if let venue = data.valueForKey("venue") as? String {
            self.venue = venue
        }
        if let memberprice = data.valueForKey("memberprice") as? String {
            self.memberprice = memberprice
        }
        if let guestprice = data.valueForKey("guestprice") as? String {
            self.guestprice = guestprice
        }
        if let date = data.valueForKey("date") as? String {
            self.date = date
        }
        if let starttime = data.valueForKey("starttime") as? String {
            self.starttime = starttime
        }
        if let endtime = data.valueForKey("endtime") as? String {
            self.endtime = endtime
        }
        if let verse = data.valueForKey("verse") as? String {
            self.verse = verse
        }
        if let details = data.valueForKey("details") as? String {
            self.details = details
        }
        if let image = data.valueForKey("image") as? String {
            self.image = image
        }
        if let form = data.valueForKey("form") as? String {
            self.form = form
        }
    }
}
